PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -766,7 +766,9 @@ COLORREF getCtrlBgColor(HWND hWnd)
 
 generic_string stringToUpper(generic_string strToConvert)
 {
-    std::transform(strToConvert.begin(), strToConvert.end(), strToConvert.begin(), ::toupper);
+    std::transform(strToConvert.begin(), strToConvert.end(), strToConvert.begin(), 
+        [](TCHAR ch){ return static_cast<TCHAR>(::toupper(ch)); }
+    );

```

Comment:
```
Warning	C4244	'=': conversion from 'int' to 'wchar_t', possible loss of data
```

Reply:
See http://en.cppreference.com/w/cpp/string/byte/toupper why lamba expression is used.

ch  
character to be converted. If the value of ch is not representable as unsigned char and does not equal EOF, the behavior is undefined.

, but TCHAR is not just a char and therefore maybe 

http://en.cppreference.com/w/cpp/string/wide/towupper

matches better here.

Reply:
Good point! MSDN points out `_totupper` exists

https://msdn.microsoft.com/pl-pl/library/45119yx3.aspx

in which case it would fit this code best. Correcting.